plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -757,6 +772,142 @@ public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTable
+                .setTableName(viewName.getTableName())
+                .setOwner(session.getUser())
+                .setTableType(org.apache.hadoop.hive.metastore.TableType.VIRTUAL_VIEW.name())
+                .setDataColumns(ImmutableList.of(dummyColumn))
+                .setPartitionColumns(ImmutableList.of())
+                .setParameters(properties)
+                .setViewOriginalText(Optional.of(encodeViewData(definition)))
+                .setViewExpandedText(Optional.of("/* Presto View */"));
+
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(VIEW_STORAGE_FORMAT)
+                .setLocation("");
+        Table table = tableBuilder.build();
+        PrincipalPrivileges principalPrivileges = buildInitialPrivilegeSet(session.getUser());
+
+        Optional<Table> existing = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName());
+        if (existing.isPresent()) {
+            if (!replace || !isPrestoView(existing.get())) {
+                throw new ViewAlreadyExistsException(viewName);
+            }
+
+            metastore.replaceTable(identity, viewName.getSchemaName(), viewName.getTableName(), table, principalPrivileges);
+            return;
+        }
+
+        try {
+            metastore.createTable(identity, table, principalPrivileges);
+        }
+        catch (TableAlreadyExistsException e) {
+            throw new ViewAlreadyExistsException(e.getTableName());
+        }
+    }
+
+    @Override
+    public void renameView(ConnectorSession session, SchemaTableName source, SchemaTableName target)
+    {
+        metastore.renameTable(new HiveIdentity(session), source.getSchemaName(), source.getTableName(), target.getSchemaName(), target.getTableName());
+    }
+
+    @Override
+    public void setViewAuthorization(ConnectorSession session, SchemaTableName viewName, TrinoPrincipal principal)
+    {
+        setTableAuthorization(session, viewName, principal);
+    }
+
+    @Override
+    public void dropView(ConnectorSession session, SchemaTableName viewName)
+    {
+        if (getView(session, viewName).isEmpty()) {
+            throw new ViewNotFoundException(viewName);
+        }
+
+        try {
+            metastore.dropTable(new HiveIdentity(session), viewName.getSchemaName(), viewName.getTableName(), true);
+        }
+        catch (TableNotFoundException e) {
+            throw new ViewNotFoundException(e.getTableName());
+        }
+    }
+
+    @Override
+    public List<SchemaTableName> listViews(ConnectorSession session, Optional<String> schemaName)
+    {
+        // Filter on ICEBERG_VIEW_COMMENT is used to avoid listing hive views in case of a shared HMS and to distinguish from materialized views
+        return schemaName.map(Collections::singletonList)
+                .orElseGet(metastore::getAllDatabases).stream()
+                .flatMap(schema -> metastore.getTablesWithParameter(schema, TABLE_COMMENT, ICEBERG_VIEW_COMMENT).stream()
```

Comment:
I think we should filter by `PRESTO_VIEW_FLAG, "true"`

(obvious, if we assume view interop between hive & Iceberg; if we do not -- let's discuss)

Reply:
Also, there is not equivalent filtering on hive side (eg HiveMetastore), is there?

Reply:
Makes sense. Let's keep it symetric and list both iceberg and hive views.

Reply:
I think  filtering on `TABLE_COMMENT, ICEBERG_VIEW_COMMENT` makes sense still (I renamed `ICEBERG_VIEW_COMMENT` to `TRINO_VIEW_COMMENT` to avoid confusion that it is something Iceberg specific).
If we filtered out on `PRESTO_VIEW_FLAG, "true" we would get materialized views here.

Reply:
Perhaps we can filter by both (`getTablesWithParameter(String,String)` -> `getTablesWithParameters(Map)`).

The underlying call (`get_table_names_by_filter`) seems to allow that.

Reply:
It does. I opt for a followup though.

Reply:
Fair.